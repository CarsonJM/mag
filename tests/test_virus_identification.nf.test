nextflow_pipeline {

    name "Test pipeline: NFCORE_MAG"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_mag"
    tag "nfcore_mag_test_virus_identification"

    test("Parameters: run_virus_identification = true") {
        when {
            params {
                outdir                  = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${params.outdir}/Assembly/MEGAHIT/MEGAHIT-group-0.contigs.fa.gz"),
                    path("${params.outdir}/multiqc/multiqc_data/multiqc_fastqc.yaml"),
                    path("${params.outdir}/VirusIdentification/"),
                ).match() },
                { assert new File("$outputDir/QC_shortreads/fastqc/test_minigut_run0_raw_1_fastqc.html").exists() },
                { assert new File("$outputDir/QC_shortreads/fastqc/test_minigut_run0_raw_2_fastqc.html").exists() },
                { assert new File("$outputDir/QC_shortreads/fastqc/test_minigut_sample2_run0_raw_1_fastqc.html").exists() },
                { assert new File("$outputDir/QC_shortreads/fastqc/test_minigut_sample2_run0_raw_1_fastqc.html").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() }
            )
        }
    }
}
