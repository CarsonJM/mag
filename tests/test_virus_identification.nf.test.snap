{
    "Parameters: run_virus_identification = true": {
        "content": [
            "MEGAHIT-group-0.contigs.fa.gz:md5,da40ccf271166ccfc233e70adb0a23ae",
            "multiqc_fastqc.yaml:md5,2ef2c5154a509b13d9bcfe8cc879d71b",
            [
                [
                    [
                        [
                            "MEGAHIT-group-0.contigs_aggregated_classification.tsv:md5,afcdcac7a8f02544c4888533e870c4e8"
                        ],
                        [
                            "MEGAHIT-group-0.contigs_taxonomy.tsv:md5,08148a4884b69088a8955c3fd8860027"
                        ],
                        [
                            "MEGAHIT-group-0.contigs_provirus.tsv:md5,93a6bca59b0bf7f57c0b9b60d2e57082"
                        ],
                        [
                            "MEGAHIT-group-0.contigs_plasmid.fna:md5,b1105afa94c08cd95e643e1534b57cd1",
                            "MEGAHIT-group-0.contigs_plasmid_genes.tsv:md5,8af86c89056190cbbe326e50b19f6bea",
                            "MEGAHIT-group-0.contigs_plasmid_proteins.faa:md5,bdd43fd82f22633079ee71dfc47ae0a1",
                            "MEGAHIT-group-0.contigs_plasmid_summary.tsv:md5,493372e1e756edb8c63318de474dacd4",
                            "MEGAHIT-group-0.contigs_virus.fna:md5,228873166621bd0c304938dd868757e2",
                            "MEGAHIT-group-0.contigs_virus_genes.tsv:md5,5bddb3bfe0a241796b26326478f06255",
                            "MEGAHIT-group-0.contigs_virus_proteins.faa:md5,4617c2217be860dbda882c6ae66a264e",
                            "MEGAHIT-group-0.contigs_virus_summary.tsv:md5,4812eb99f0e3b162ae1b8a747d54e9fe"
                        ]
                    ]
                ]
            ]
        ],
        "timestamp": "2023-12-07T11:20:44.005938987"
    }
}